pkg/resource/replication_group/custom_update_api.go

```diff
@@ -515,6 +515,13 @@ func (rm *resourceManager) newUpdateShardConfigurationRequestPayload(
 			res.SetReshardingConfiguration(shardsConfig)
 		}
 	} else if decrease {
+		// Compute shards to retain
+		if desiredSpec.NodeGroupConfiguration == nil {
+			for i := 0;  i < int(*desiredShardsCount); i++ {
+				shardsToRetain = append(shardsToRetain, desired.ko.Status.NodeGroups[i].NodeGroupID)
+			}
+		}
```

Comment:
Please consider moving this logic to an `else` block at line number 497. `shardsToRetain` is populated during if block and the else block would be logically connected to it.